business/checkers/virtual_services/route_checker.go

```diff
@@ -95,13 +95,12 @@ func (route RouteChecker) checkRoutesFor(kind string) ([]*models.IstioCheck, boo
 			path := fmt.Sprintf("spec/%s[%d]/route", kind, routeIdx)
 			validation := buildValidation("Weight sum should be 100", "error", path)
 			validations = append(validations, &validation)
-		}
-
-		if weightCount > 0 && weightCount != destinationWeights.Len() {
-			valid = false
-			path := fmt.Sprintf("spec/%s[%d]/route", kind, routeIdx)
-			validation := buildValidation("All routes should have weight", "error", path)
-			validations = append(validations, &validation)
+			if weightCount != destinationWeights.Len() {
+				valid = false
+				path := fmt.Sprintf("spec/%s[%d]/route", kind, routeIdx)
+				validation := buildValidation("All routes should have weight", "error", path)
```

Comment:
@abonas I thought exactly the same as you, but if you expand the lines right above the change you'll see that the indentation is correct. That code is inside an `if-else` statement now.

@amyroh thanks for the contribution :) 
In understand that with this changes, we show an error validations when:
* sum weight is between 0 and 100
* AND there are more than 1 routes without weight.

Here we have how the validations are displayed:
![screenshot of kiali console 15](https://user-images.githubusercontent.com/613814/50398456-056cd800-0778-11e9-888b-c9abfe64c7f4.png)

In case all the routes have weight, then we will only show the one validation:
![screenshot of kiali console 16](https://user-images.githubusercontent.com/613814/50398814-d146e680-077a-11e9-92e2-4ab6d233cb84.png)

This looks good to me. Perhaps we could move down the severity to warning for the "All routes should have weight". Just an idea, wdyt?

Reply:
Sounds good!  Thanks for reviewing.